### PR TITLE
Follow hlint suggestion: move brackets to avoid $

### PR DESCRIPTION
**Pull request by philderbeast**
See #9110.

### CODE REVIEW

File: Cabal/src/Distribution/Simple/Haddock.hs [hunk 1]
```diff
@@ -395,26 +395,28 @@ haddock pkg_descr lbi suffixes flags' = do
...
-                let libArgs' = commonArgs `mappend` flibArgs
-                runHaddock verbosity tmpFileOpts comp platform haddockProg True libArgs'
-        )
+        when
+          (flag haddockForeignLibs)
+          ( do
+              withTempDirectoryEx verbosity tmpFileOpts (buildDir lbi') "tmp" $
+                \tmp -> do
+                  smsg
+                  flibArgs <-
```
Comment (ffaf1) [inline Cabal/src/Distribution/Simple/Haddock.hs:404]: I do not fancy this one

File: cabal-install/src/Distribution/Client/List.hs [hunk 2]
```diff
@@ -466,8 +464,7 @@ showPackageDetailedInfo pkginfo =
         <<>> maybe Disp.empty (\v -> char '-' Disp.<> pretty v) (selectedVersion pkginfo)
       <+> text (replicate (16 - length (prettyShow (pkgName pkginfo))) ' ')
         <<>> parens pkgkind
-      $+$ ( nest 4 $
-              vcat
+      $+$ nest 4 (vcat
```
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/List.hs:467]: Closing bracket 30+ lines later, odd but followable.

File: cabal-install/src/Distribution/Client/ProjectPlanOutput.hs [hunk 3]
```diff
@@ -193,7 +193,8 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
               let components =
                     J.object $
                       [ comp2str c
-                        J..= ( J.object $
+                        J..= J.object
+                             (
```
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/ProjectPlanOutput.hs:197]: Again a weird Allman-style bracket, but still readable in an editor.

**Review by ffaf1 — APPROVED** [review]
Apart from the `do` block, reasonable.

File: Cabal/src/Distribution/Simple/Haddock.hs (hunk 1 above)
Comment (philderbeast) [reply Cabal/src/Distribution/Simple/Haddock.hs:404]: I get that. I looked at it a while and concluded the difficulty in reading the code block was there beforehand.

File: cabal-install/src/Distribution/Client/List.hs (hunk 2 above)
Comment (philderbeast) [reply cabal-install/src/Distribution/Client/List.hs:467]: I could have used indentation but didn't want to unindent so many lines, aiming for a small change.

File: Cabal/src/Distribution/Simple/Haddock.hs (hunk 1 above)
Comment (andreabedini) [reply Cabal/src/Distribution/Simple/Haddock.hs:404]: I agree, perhaps you should consider trying to make this part it more readable.

File: cabal-install/src/Distribution/Client/List.hs (hunk 2 above)
Comment (andreabedini) [reply cabal-install/src/Distribution/Client/List.hs:467]: The closing `)`  was also far before right?

**Review by andreabedini — APPROVED** [review]
I am good with this :+1:

File: Cabal/src/Distribution/Simple/Haddock.hs (hunk 1 above)
Comment (ulysses4ever) [reply Cabal/src/Distribution/Simple/Haddock.hs:404]: Personally, I'm a heavy user of `$` and I also don't like these new parens. But I can live with them no problem.

File: cabal-install/src/Distribution/Client/List.hs (hunk 2 above)
Comment (ulysses4ever) [reply cabal-install/src/Distribution/Client/List.hs:467]: Yes, +1 to Andrea's comment: it was far away in the initial version too.

File: cabal-install/src/Distribution/Client/ProjectPlanOutput.hs (hunk 3 above)
Comment (ulysses4ever) [reply cabal-install/src/Distribution/Client/ProjectPlanOutput.hs:197]: Not sure who's Allman (Wikipedia says, his indentation style does 8 spaces for a tab, and that's about it) but an extra line  that the new version takes is unfortunate, indeed. Again, hardly critical.

**Review by ulysses4ever — APPROVED** [review]
Don't see any controversy here. Thanks!

**Pull request merged**